save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/OrganizationController.kt

```diff
@@ -347,14 +345,56 @@ internal class OrganizationController(
             "Not enough permission for deletion of organization $organizationName."
         }
         .filter {
-            !organizationService.hasProjects(organizationName)
+            !organizationService.hasProjects(organizationName) || status == OrganizationStatus.BANNED
         }
         .switchIfEmptyToResponseException(HttpStatus.CONFLICT) {
             "There are projects connected to $organizationName. Please delete all of them and try again."
         }
         .map {
-            organizationService.deleteOrganization(it.name)
-            ResponseEntity.ok("Organization deleted")
+            if (status == OrganizationStatus.DELETED) {
+                organizationService.deleteOrganization(it.name)
+            } else if (status == OrganizationStatus.BANNED) {
+                organizationService.banOrganization(it.name)
+            }
+            ResponseEntity.ok("Organization deleted oa banned")
+        }
+
+    @PostMapping("/{organizationName}/recover")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("isAuthenticated()")
+    @Operation(
+        method = "POST",
+        summary = "Recover existing organization.",
+        description = "Recover existing organization by its name.",
+    )
+    @Parameters(
+        Parameter(name = "organizationName", `in` = ParameterIn.PATH, description = "name of an organization", required = true),
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully recovered an organization.")
+    @ApiResponse(responseCode = "403", description = "Not enough permission for recovering this organization.")
+    @ApiResponse(responseCode = "404", description = "Could not find deleted organization with such name.")
+    fun recoverOrganization(
+        @PathVariable organizationName: String,
+        authentication: Authentication,
+    ): Mono<StringResponse> = Mono.just(organizationName)
+        .flatMap {
```

Comment:
you call `.flatMap` on operation where you call `toMono()`. I guess you did it to avoid nullable organization, but I suggest to change it to:
* `{}.toMono()`
```kotlin
{
    organizationService.findByName(organizationName)
}
.toMono()
.filter {
    it.status != OrganizationStatus.CREATED
}
```
* `blockingToMono {}`
```kotlin
blockingToMono {
    organizationService.findByName(organizationName)
}
.filter {
    it.status != OrganizationStatus.CREATED
}
```

P.S. `blockingToMono` is preferable since it uses proper Scheduler for blocking operations